src/coordinate_search.jl

```diff
@@ -0,0 +1,65 @@
+export coordinate_search
+
+"""
+	The coordinate search is a derivative free optimization method.
+	It minimizes a certain function f by walking on a grid and taking a step where the function values are smaller.
+	When it can no longer find any smaller values, it reduces the grid by a factor of 2, repeating the process.
+
+"""
+
```

Comment:
No space here, this is a docstring, it should be put right before the function definition. You can test it's working with `?coordinate_search`. It should follow the syntax:
```
"""
    coordinate_search(nlp; options...)

Blah blah, where `nlp` is an `AbstractNLPModel`. The following options are available:
- `tol::Real = 1e-8`
- ...
"""
```